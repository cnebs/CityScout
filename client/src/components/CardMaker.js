import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import AddCircle from '@material-ui/icons/AddCircle';
import AddIcon from '@material-ui/icons/Add';
import CalendarIcon from '@material-ui/icons/CalendarToday';
import Button from '@material-ui/core/Button';
import { Grow, Fab } from '@material-ui/core/';
import moment from 'moment';
import Link from '@material-ui/core/Link';


const useStyles = makeStyles(theme => ({
  card: {
    maxWidth: 360
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

export default function CardMaker({ event, animationTime, handleAddToCalClick }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  function handleExpandClick() {
    setExpanded(!expanded);
  }

  return (
    <Grow in={true} timeout={animationTime}>
      <Card className={classes.card}>

        {/* <CardHeader
        // Stuff above the image would go here
        /> */}
        <CardMedia
          className={classes.media}
          image={event.img}
        />

        <CardContent>
          <Typography variant="h6"  >
            <Link href={event.url} color="textPrimary">
              {event.name}
            </Link>
          </Typography>
          <Typography variant="body2" color="textSecondary">
            {moment(event.time_start).format('ddd, MMM DD, h:mm a')}
          </Typography>
          <Typography variant="body2" color="textSecondary">
            {event.venue}, {event.location}
          </Typography>
        </CardContent>

        <CardActions disableSpacing>
          <Fab
            color="secondary"
            aria-label="add"
            onClick={() => { handleAddToCalClick(event) }}
          >
            {/* <AddIcon /> */}
            <CalendarIcon />
          </Fab>

          <IconButton
            className={clsx(classes.expand, {
              [classes.expandOpen]: expanded,
            })}
            onClick={handleExpandClick}
            aria-expanded={expanded}
            aria-label="show more"
          >
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>

        {/* Collapse Section */}

        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Typography paragraph>
              Description? {event.description}
            </Typography>
          </CardContent>
        </Collapse>

      </Card>
    </Grow>
  );
}
