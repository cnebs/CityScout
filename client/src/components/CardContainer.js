import React from 'react';
import CardMaker from './CardMaker.js';
import { Grid, Typography } from '@material-ui/core/'
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  cardColumn: {
    padding: 20
  }
}));

export default function CardContainer({ event, day, animationTime, handleCardActionClick }) {
  const classes = useStyles();

  return (
    <>
      <Grid align='center' item xs={12} md={4} className={classes.cardColumn}>
        <Typography variant='h5' align='center' gutterBottom={true} color="textSecondary">
          {day}
        </Typography>
        <CardMaker
          event={event}
          animationTime={animationTime}
          handleCardActionClick={handleCardActionClick} />
      </Grid>
    </>
  );
}