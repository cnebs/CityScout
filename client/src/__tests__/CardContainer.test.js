import { shallow } from 'enzyme';
import React from 'react';
import CardContainer from '../components/CardContainer';
import CardMaker from '../components/CardMaker';
import { Grid, Typography } from '@material-ui/core/';

describe('CardMaker component', () => {
  let wrapper;
  let event = {
    source_API: 'TicketMaster',
    name: 'Hayes Carll',
    url:
      'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2718472',
    event_id: 'Z7r9jZ1Aejbow',
    time_start: '2019-08-11T02:00:00Z',
    time_end: null,
    category: 'Music',
    image:
      'https://s1.ticketm.net/dam/a/fc1/e7affb5a-4ba1-4e6f-8aad-29c79f4a6fc1_68981_RECOMENDATION_16_9.jpg',
    venue: 'Gruene Hall',
    location: 'New Braunfels',
    price_min: null,
    price_max: null,
    description: null
  };
  let day = 'testing day';

  beforeAll(() => {
    wrapper = shallow(<CardContainer event={event} day={day} />);
  });

  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders the correct child components', () => {
    expect(wrapper.find(CardMaker).length).toBe(1); // should hold only 1 card
    expect(wrapper.find(Typography).length).toBeGreaterThan(0);
    expect(wrapper.find(Grid).length).toBeGreaterThan(0);
  });
});
