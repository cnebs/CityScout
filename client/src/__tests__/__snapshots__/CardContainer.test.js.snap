// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMaker component it renders without crashing 1`] = `ShallowWrapper {}`;
